Modules/_csv.c

```diff
@@ -1171,10 +1169,14 @@ csv_writerow(WriterObj *self, PyObject *seq)
     PyObject *iter, *field, *line, *result;
 
     iter = PyObject_GetIter(seq);
-    if (iter == NULL)
-        return PyErr_Format(_csvstate_global->error_obj,
-                            "iterable expected, not %.200s",
-                            Py_TYPE(seq)->tp_name);
+    if (iter == NULL) {
+        if (PyErr_ExceptionMatches(PyExc_TypeError)) {
+            PyErr_Format(_csvstate_global->error_obj,
+                         "iterable expected, not %.200s",
+                         Py_TYPE(seq)->tp_name);
+        }
+        return NULL;
```

Comment:
Is it possible for `PyObject_GetIter(seq)` to return NULL without setting an exception? If so, we may want to consider checking `PyErr_Occurred()` before `PyErr_ExceptionMatches(PyExc_TypeError)`.

Reply:
It should not. If it happens due to a programming error, it will likely lead to a crash.